optimum/onnx/modeling_seq2seq.py

```diff
@@ -0,0 +1,59 @@
+#  Copyright 2022 The HuggingFace Team. All rights reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+from typing import Optional, Tuple
+
+import torch
+from transformers import PreTrainedModel
+
+
+# Currently inherits from PreTrainedModel for export constraint coming from transformers.onnx.export
```

Comment:
I was making reference to the export being only applied on [PreTrainedModels](https://github.com/huggingface/transformers/blob/main/src/transformers/onnx/convert.py#L334). Do you think we could we extend this constraint to torch.nn.Module ? We can otherwise have our own export function in optimum but that comes with the cost of duplicating code which is not a very nice solution.